apple/testing/default_runner/ios_xctestrun_runner.template.sh

```diff
@@ -248,13 +235,52 @@ if [[ $(arch) == arm64 && "$test_file" != *arm64* ]]; then
   intel_simulator_hack=true
 fi
 
+should_use_xcodebuild=false
+if [[ -n "$test_host_path" ]]; then
+  echo "note: Using 'xcodebuild' because test host was provided"
+  should_use_xcodebuild=true
+fi
 # shellcheck disable=SC2050
-if [[ -n "$test_host_path" || -n "${CREATE_XCRESULT_BUNDLE:-}" || "%(test_order)s" == random ]]; then
+if [[ "%(test_order)s" == random ]]; then
+  echo "note: Using 'xcodebuild' because random test order was requested"
+  should_use_xcodebuild=true
+fi
+if [[ -n "${CREATE_XCRESULT_BUNDLE:-}" ]]; then
+  echo "note: Using 'xcodebuild' because XCResult bundle was requested"
+  should_use_xcodebuild=true
+fi
+if [[ -n "$xctestrun_skip_test_section" || -n "$xctestrun_only_test_section" ]]; then
+  echo "note: Using 'xcodebuild' because test filter was provided"
```

Comment:
i think with this one we probably could support the other option but this is probably good for now